src/CoreWCF.Http/src/CoreWCF/Channels/HttpChannelHelpers.cs

```diff
@@ -401,30 +352,27 @@ public async Task<(Message message, Exception requestException)> ParseIncomingMe
                     }
                     else
                     {
-                        return (null, requestException);
+                        return (null, null, null);
                     }
                 }
                 else
                 {
                     Stream stream = GetInputStream(true);
+                    reader = PipeReader.Create(stream, new StreamPipeReaderOptions(bufferSize: ConnectionOrientedTransportDefaults.ConnectionBufferSize));
                     if (_streamed)
```

Comment:
The buffer size should be ultimately coming from HttpTransportBindingElement.MaxBufferSize. I can't remember if that's on the settings class already, I think it might be.

Reply:
@mconnew the PipeReaderOptions allows to configure a min buffer size but not a max buffer size. https://learn.microsoft.com/en-us/dotnet/api/system.io.pipelines.streampipereaderoptions.buffersize?view=dotnet-plat-ext-7.0#system-io-pipelines-streampipereaderoptions-buffersize

Reply:
MemoryPool<T> does allow to configure a max buffer size and it is possible to specify a custom pool in PipeReaderOptions. https://learn.microsoft.com/en-us/dotnet/api/system.buffers.memorypool-1?view=net-7.0&viewFallbackFrom=dotnet-plat-ext-7.0. 
But once we move to .NET and use BodyReader property I d'ont think we will be able to specify the pool.